[WIP] Refactor FlashCLI

karthikrangasai opened the pull request:
## What does this PR do?

<!--
Please include a summary of the change and which issue is fixed.
 Please also include relevant motivation and context.
 List any dependencies that are required for this change.

Fixes # (issue)
-->

The `LightningCLI` class has now been merged on Lightning master and we can import it from Lightning directly.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

<!-- For CHANGELOG separate each item in unreleased section by a blank line to reduce collisions -->

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

karthikrangasai commented:
Hello everyone,

Do we need to add a testing for `FlashCLI` like `LightningCLI` is tested in Lightning repo ?

SeanNaren commented:
hey @karthikrangasai thanks for the PR! 

It would be nice to have a file containing tests for functionality that is only available in the FlashCLI! Would you be able to set this up? things such as https://github.com/PyTorchLightning/lightning-flash/blob/569a2c705a8efabdca645ef630644d795a026190/flash/core/utilities/flash_cli.py#L222 or https://github.com/PyTorchLightning/lightning-flash/blob/569a2c705a8efabdca645ef630644d795a026190/flash/core/utilities/flash_cli.py#L226 should be easy to test :)

karthikrangasai commented:
Hello @SeanNaren ,

Sure, I will work on it and update the PR.

carmocca commented:
Most of the code in the `FlashCLI` should also be removed. I can help here with any questions.

What's important is to have tests for the expected integration and then we can have confidence in the replacement.

karthikrangasai commented:
Hello @carmocca ,

Sorry I had missed your comments yesterday. I have added the latest code from the Lighting master branch as the one I had deleted was older version. I have also updated the minimum `jsonargparse` version.

Yes, I would like some clarity on the changes to be made in the `FlashCLI` code.

carmocca commented on flash/core/utilities/imports.py:
```diff
@@ -102,6 +102,7 @@ def _compare_version(package: str, op, version) -> bool:
 _VISSL_AVAILABLE = _module_available("vissl") and _module_available("classy_vision")
 _ALBUMENTATIONS_AVAILABLE = _module_available("albumentations")
 _BAAL_AVAILABLE = _module_available("baal")
+_JSONARGPARSE_AVAILABLE = _module_available("jsonargparse")
```
Flash installs jsonargparse unconditionally so this shouldn't be necessary

carmocca reviewed (commented):
> Yes, I would like some clarity on the changes to be made in the FlashCLI code.

Basically most of the code should be removed but the init arguments kept. It's hard for me to say because I need to get more familiar with what the `FlashCLI` tries to do and because there is a lot to remove :P

Can you gather a list of working examples and/or tests so that we can validate the changes? I'm unsure about the test coverage on this, especially across tasks

karthikrangasai commented:
not stale
